icx-proxy/src/config/canister_dns_config.rs

```diff
@@ -11,9 +12,11 @@ impl CanisterDnsConfig {
     /// Create a CanisterDnsConfig instance from command-line configuration.
     /// dns_aliases: 0 or more entries of the form of dns.alias:canister-id
     pub fn new(dns_aliases: &[String]) -> anyhow::Result<CanisterDnsConfig> {
+        let mut dns_aliases: Vec<String> = dns_aliases.iter().map(String::clone).collect();
+        dns_aliases.sort_by_key(|s| Reverse(s.len()));
         let dns_aliases = dns_aliases
             .iter()
-            .map(|alias| DnsAlias::new(alias))
+            .map(|alias| DnsAlias::new(&alias))
             .collect::<anyhow::Result<Vec<DnsAlias>>>()?;
```

Comment:
I with we wouldn't have to collect twice, but oh well... It's just at startup anyway.

Reply:
Me too, but I couldn't sort in between `.iter()` and `.collect()`..  I have an idea though